contracts/strategies/lift/LiftStrategy.sol

```diff
@@ -0,0 +1,272 @@
+  *   Note that they cannot come in take away coins that are used and defined in the strategy itself
+  *   Those are protected by the "unsalvagableTokens". To check, see where those are being flagged.
+  */
+  function salvage(address recipient, address token, uint256 amount) external onlyControllerOrGovernance {
+     // To make sure that governance cannot come in and take away the coins
+    require(!unsalvagableTokens[token], "token is defined as not salvagable");
+    IERC20(token).safeTransfer(recipient, amount);
+  }
+
+  /*
+  *   Get the reward, sell it in exchange for underlying, invest what you got.
+  *   It's not much, but it's honest work.
+  *
+  *   Note that although `onlyNotPausedInvesting` is not added here,
+  *   calling `investAllUnderlying()` affectively blocks the usage of `doHardWork`
+  *   when the investing is being paused by governance.
+  */
+  function doHardWork() external onlyNotPausedInvesting restricted {
+    uint256 rewardBalanceBefore = boardRoom.getbalanceOfShare(address(this));
+    rewardPool.stakeInBoardroom();
+    uint256 rewardBalanceAfter = boardRoom.getbalanceOfShare(address(this));
+    uint256 rewardStaked = rewardBalanceAfter.sub(rewardBalanceBefore);
+    if (rewardStaked > 0) {
+      stakes.push([block.timestamp, rewardStaked]);
+    }
+    investAllUnderlying();
+  }
+
+  function _getReward() internal {
+    uint256 rewardBalanceBefore = boardRoom.getbalanceOfShare(address(this));
+    rewardPool.stakeInBoardroom();
+    uint256 rewardBalanceAfter = boardRoom.getbalanceOfShare(address(this));
+    uint256 rewardStaked = rewardBalanceAfter.sub(rewardBalanceBefore);
+    if (rewardStaked > 0) {
+      stakes.push([block.timestamp, rewardStaked]);
+    }
+  }
+
+  function withdrawRewardShareOldest() external onlyMultiSigOrGovernance {
+    require(allowedRewardClaimable, "reward claimable is not allowed");
+    uint256 len = stakes.length;
+    uint256 shareBalance = IERC20(rewardToken).balanceOf(address(this));
+    uint256 i = 0;
+    for (i;i<len;i++) {
+      uint256 time = stakes[i][0];
+      uint256 amount = stakes[i][1];
+      if (amount > 0) {
+        boardRoom.withdrawShare(time);
+        shareBalance = IERC20(rewardToken).balanceOf(address(this));
+        stakes[i] = [0,0];
+        break;
+      }
+    }
+    IERC20(rewardToken).safeTransfer(msg.sender, shareBalance);
+  }
+
+  function withdrawRewardShareNewest() external onlyMultiSigOrGovernance {
+    require(allowedRewardClaimable, "reward claimable is not allowed");
+    uint256 len = stakes.length;
+    uint256 shareBalance = IERC20(rewardToken).balanceOf(address(this));
+    uint256 i = len-1;
+    for (i;i>=0;i--) {
+      uint256 time = stakes[i][0];
+      uint256 amount = stakes[i][1];
+      if (amount > 0) {
+        boardRoom.withdrawShare(time);
+        shareBalance = IERC20(rewardToken).balanceOf(address(this));
+        stakes[i] = [0,0];
```

Comment:
any reason we are not doing .pop() here?

Reply:
I mean, is the intention to always zero an item after calling `boardRoom.withdrawShare(...)` on it? If yes I would suggest instead of clearing an item, just removing it. Therefore, the first non-zero item (from the end) will be the one that we are interested in, and we no longer need to have a for loop.

Reply:
I was actually thinking of doing this earlier. Will update.

Reply:
Fixed